provers/sp1/driver/src/proof_verify/remote_contract_verify.rs

```diff
@@ -0,0 +1,57 @@
+use alloy_primitives::Address;
+use alloy_provider::ProviderBuilder;
+use alloy_sol_types::sol;
+use raiko_lib::prover::ProverResult;
+use reth_primitives::B256;
+use sp1_sdk::proto::network::twirp::url::Url;
+use std::{env, str::FromStr};
+use tracing::{error, info};
+
+use crate::RaikoProofFixture;
+
+sol!(
+    #[sol(rpc)]
+    #[allow(dead_code)]
+    contract ISP1Verifier {
+        #[derive(Debug)]
+        function verifyProof(
+            bytes32 programVKey,
+            bytes calldata publicValues,
+            bytes calldata proofBytes
+        ) external view;
+    }
+);
+
+/// using pre-deployed contract to verify the proof, the only problem is to double check the verification version.
+pub(crate) async fn verify_sol_by_contract_call(fixture: &RaikoProofFixture) -> ProverResult<()> {
+    let sp1_verifier_rpc_url = env::var("SP1_VERIFIER_RPC_URL").expect("env SP1_VERIFIER_RPC_URL");
+    let sp1_verifier_addr = {
+        let addr = env::var("SP1_VERIFIER_ADDRESS").expect("env SP1_VERIFIER_RPC_URL");
+        Address::from_str(&addr).unwrap()
+    };
```

Comment:
Maybe this should also look at the verifier address set in the config?

Reply:
not the same one, the one in config is Taiko's tier verifier which has extra logic for publicInput & some image id checkings, this one is pure plonk verifier from sp1. The use case is if a proof passes here, it should pass our tier verifier unless the public input is incorrect (should be the only exception).  We will use it to do the stress test (that 1 week stressing) in mainnet, as mainnet does not have sp1 verifier, so we know if any proof is fail.
A future refactory is using sp1's official verifier address, then we can hardcode it in raiko to avoid reading env here.